src/internal_rules.jl

```diff
@@ -742,11 +758,9 @@ function EnzymeRules.augmented_primal(
     RT::Type,
     A::Annotation{<:Union{Matrix,LinearAlgebra.RealHermSym{<:Real,<:Matrix}}};
     kwargs...)
-    fact = if EnzymeRules.needs_primal(config)
-        cholesky(A.val; kwargs...)
-    else
-        nothing
-    end
+
+    fact = cholesky(A.val; kwargs...)
```

Comment:
This should be `(EnzymeRules.needs_primal(config) || !(RT <: Const)) ? cholesky(...) : nothing` (aka only creating it if we would use it